nautobot/extras/managers.py

```diff
@@ -89,6 +89,10 @@ def store_result(
             if fields["status"] in states.READY_STATES:
                 fields["date_done"] = timezone.now()
 
+            # Don't overload the user if it's already been set when the JobResult was created.
+            if obj.user_id is not None:
+                del fields["user_id"]
```

Comment:
@gsnider2195 This was my interim fix for `JobResult.user` getting nullified. You fixed this with the context changes, right?

Reply:
Yes the fix was to manually set `user` on the jobresult after running `apply()` [here](https://github.com/nautobot/nautobot/blob/e47608214277dcc4550e2e32c4389807c183fb2e/nautobot/extras/models/jobs.py#L763-L790)

Reply:
Hmm I think these are different problems, because yours is for eager results, and the issue I encountered was during normal job runs (async). I'll pull this out and see whether it's still necessary.

Reply:
If you're seeing it during normal job runs you need to make sure you're setting the `user_id` kwargs if you're calling `apply_async` manually instead of using `enqueue_job`

Reply:
This should be removed before merging unless you're still seeing problems related to the user being removed from the job result